olive/model/handler/hf.py

```diff
@@ -0,0 +1,213 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import logging
+from pathlib import Path
+from typing import Any, ClassVar, Dict, List, Optional, Tuple, Union
+
+import torch
+
+from olive.common.config_utils import serialize_to_json, validate_config
+from olive.common.utils import dict_diff
+from olive.constants import Framework
+from olive.hardware.accelerator import Device
+from olive.model.config import HfLoadKwargs, IoConfig
+from olive.model.config.registry import model_handler_registry
+from olive.model.handler.base import OliveModelHandler
+from olive.model.handler.mixin import HfMixin, MLFlowMixin
+from olive.model.handler.pytorch import PyTorchModelHandlerBase
+from olive.model.utils.hf_utils import DEFAULT_HF_TASK, load_hf_model_from_task
+from olive.resource_path import OLIVE_RESOURCE_ANNOTATIONS
+
+logger = logging.getLogger(__name__)
+
+
+@model_handler_registry("HFModel")
+class HfModelHandler(PyTorchModelHandlerBase, MLFlowMixin, HfMixin):  # pylint: disable=too-many-ancestors
+    resource_keys: Tuple[str, ...] = ("model_path", "adapter_path")
+    json_config_keys: Tuple[str, ...] = ("task", "load_kwargs", "generative")
+
+    def __init__(
+        self,
+        model_path: OLIVE_RESOURCE_ANNOTATIONS,
+        task: str = DEFAULT_HF_TASK,
+        load_kwargs: Union[Dict[str, Any], HfLoadKwargs] = None,
+        io_config: Union[Dict[str, Any], IoConfig, str] = None,
+        adapter_path: OLIVE_RESOURCE_ANNOTATIONS = None,
+        model_attributes: Optional[Dict[str, Any]] = None,
+        generative: bool = False,
+    ):
+        super().__init__(
+            framework=Framework.PYTORCH,
+            model_file_format=None,
+            model_path=model_path,
+            model_attributes=model_attributes,
+            io_config=io_config,
+            generative=generative,
+        )
+        self.add_resources(locals())
+        self.task = task
+        self.load_kwargs = validate_config(load_kwargs, HfLoadKwargs, warn_unused_keys=False) if load_kwargs else None
+
+        self.mlflow_model_path = None
```

Comment:
nit: mlflow is not strongly bound with transformers. Maybe update the corresponding name to `mlflow_transformers`? Then we can distinguish the normal mlflow model and mlflow-transoformers models.

Reply:
done